### PR TITLE
Init markersDispatcher_ on componentDidMount

### PR DESCRIPTION
**Pull request by bordeo**
Due to [React Strict Mode updates](https://reactjs.org/blog/2022/03/08/react-18-upgrade-guide.html#updates-to-strict-mode:~:text=To%20do%20this%2C%20React%20would%20unmount%20and%20remount%20trees%20using%20the%20same%20component%20state%20as%20before.) `markersDispatcher_` needs to be initialized on `componentDidMount`
This pull request fixes: 
- #1110 
- #1113 
- #1116 
- #1145

### CODE REVIEW

File: src/google_map.js [hunk 1]
```diff
@@ -259,6 +259,7 @@ class GoogleMap extends Component {
 
   componentDidMount() {
     this.mounted_ = true;
+    this.markersDispatcher_ = new MarkerDispatcher(this);
```
Comment (mcgrealife) [inline src/google_map.js:262]: important

File: package.json [hunk 2]
```diff
@@ -55,8 +55,8 @@
     "microbundle-crl": "^0.13.10",
     "npm-run-all": "^4.1.5",
     "prettier": "^2.0.4",
-    "react": "^16.0.0 || ^17.0.2 || ^18.0.0",
-    "react-dom": "^16.0.0 || ^17.0.2 || ^18.0.0",
+    "react": "^18.2.0",
```
Comment (mcgrealife) [inline package.json:58]: important

File: package.json [hunk 3]
```diff
@@ -55,8 +55,8 @@
...
+    "react-dom": "^18.2.0",
```
Comment (mcgrealife) [inline package.json:59]: important

File: package.json [hunk 4]
```diff
@@ -16,7 +16,7 @@
   },
   "scripts": {
     "build": "microbundle-crl -f cjs,umd src/index.cjs.js && microbundle-crl -f es,modern src/index.js",
-    "start": "microbundle-crl watch -f cjs,umd",
+    "start": "microbundle-crl watch -f es,modern src/index.js",
```
Comment (mcgrealife) [inline package.json:19]: important

**Comment by mcgrealife** [comment]
_To help future reviewers parse important lines:_

### Primary (screenshot):
1. Upgrade `react` and `react-dom`
2. Init `markersDispatcher_` on componentDidMount
3. Update package.json `start` command

important changes

### Secondary:
- Update example
- Formatting (semicolons, spacing)

**Comment by Eslam3bed** [comment]
Thanks, @bordeo for the initiative 
any updates on when this will be reviewed 👀

**Comment by wescopeland** [comment]
@itsmichaeldiego This issue is affecting many projects using React 18, with users having to either remain on React 17 or disable React strict mode. What can we do to escalate getting this PR reviewed?

**Comment by Eslam3bed** [comment]
> #### @itsmichaeldiego can we have your help on this 🙏🏼 ?

**Comment by wescopeland** [comment]
I think it's safe to assume this package has been abandoned. This issue has been occurring since the v18 RC builds all the way back in March. At my org we are running a fork that fixes the issue but has problems of its own.

**Comment by KasimAhmic** [comment]
Any rough estimate as to when this can get merged?

**Comment by edzis** [comment]
Here is my trail of issues related to React 18 if it helps anyone to follow through:

## v2.1.10:
- markers jitter after drag
- in StrictMode causes infinite crash loop

See https://www.loom.com/share/8a010b76a6504d11b6d6e3a6fae83b1a

## https://github.com/google-map-react/google-map-react/pull/1146 (this PR)
- does not fix jitter
- fixes StrictMode crash loop
- in StrictMode causes markers to move to original position after drag

See https://www.loom.com/share/2995f667d363417ba9fe4b4286e72aee

## https://github.com/bordeo/google-map-react/pull/1 on top of https://github.com/bordeo/google-map-react/pull/1:
- fixes jitter, introduces `Warning: flushSync was called from inside a lifecycle method. React cannot flush when React is already rendering. Consider moving this call to a scheduler task or micro task.`

See https://www.loom.com/share/cb6c55ae652c4b17a8efb9924a1f0ccb

---

For now I am setting on my PR and without StrictMode.

**Comment by neo0910** [comment]
Hello!
Will this PR be merged?

**Comment by itsmichaeldiego** [comment]
Merged and published in version: 2.2.1, please let me know if it works for you, thanks!

**Comment by giorgiabosello** [comment]
@itsmichaeldiego did you tried it before merging it?
http://google-map-react.github.io/google-map-react/map/main/ if you zoom or change map position, markers change with it because they are not reinitialized.

**Comment by itsmichaeldiego** [comment]
@giorgiabosello That issue is not related to this version! We don't use that map example anymore! 
Please check in your local and let me know, thanks :D

**Comment by giorgiabosello** [comment]
@itsmichaeldiego same 😅

**Pull request merged**